kafka/channel/pkg/reconciler/controller/kafkachannel.go

```diff
@@ -401,8 +401,8 @@ func (r *Reconciler) reconcileDispatcher(ctx context.Context, dispatcherNamespac
 		logging.FromContext(ctx).Error("Unable to get the dispatcher deployment", zap.Error(err))
 		kc.Status.MarkServiceUnknown("DispatcherDeploymentFailed", "Failed to get dispatcher deployment: %v", err)
 		return nil, err
-	} else if !reflect.DeepEqual(expected.Spec, d.Spec) {
-		logging.FromContext(ctx).Info("Deployment is not what we expect it to be, updating Deployment")
+	} else if !reflect.DeepEqual(expected.Spec.Template.Spec.Containers[0].Image, d.Spec.Template.Spec.Containers[0].Image) {
```

Comment:
Just noticed this going through release notes. This seems like a job for DeepDerivative.

Reply:
I could've sworn I tried that and it didn't work.

Reply:
That's certainly possible :D